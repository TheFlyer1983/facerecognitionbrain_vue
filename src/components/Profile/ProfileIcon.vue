<script setup lang="ts">
import { ref } from 'vue';

const isDropdownOpen = ref(false);

function toggleDropdown() {
  isDropdownOpen.value = !isDropdownOpen.value;
}
</script>

<template>
  <section class="dropdown">
    <span @click="toggleDropdown" class="dropdown-button">
      <img
        src="http://tachyons.io/img/logo.jpg"
        alt="avatar"
        class="dropdown-button-image"
      />
    </span>
    <ul v-if="isDropdownOpen" class="dropdown-menu">
      <li class="dropdown-menu-item">View Profile</li>
      <li class="dropdown-menu-item">Sign Out</li>
    </ul>
  </section>
</template>

<style scoped lang="scss">
.dropdown {
  position: relative;
  margin: 1rem;

  &-button {
    display: flex;
    justify-content: flex-end;
    border: none;
    font-size: inherit;

    &-image {
      border-radius: 100%;
      border: 1px solid black;
      display: inline-block;
      height: 4rem;
      width: 4rem;
    }
  }

  &-menu {
    position: absolute;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    right: 0;
    border: transparent;
    margin-top: 20px;
    background-color: rgba(255, 255, 255, 0.5);
    width: 10rem;
    list-style: none;
    padding-left: auto;
    padding-right: auto;
    padding-top: 1rem;
    border-radius: 5px;

    &-item {
      cursor: pointer;
      padding-bottom: 1rem;
    }
  }
}
</style>
