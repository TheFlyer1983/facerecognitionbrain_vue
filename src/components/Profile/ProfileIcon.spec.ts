import { shallowMount, VueWrapper } from '@vue/test-utils';
import { beforeEach, describe, it, expect, vi, beforeAll } from 'vitest';
import { reactive } from 'vue';
import { useStore } from 'vuex';
import ProfileIcon from './ProfileIcon.vue';

vi.mock('vuex');

const mockedUseStore = vi.mocked<() => Partial<typeof useStore>>(useStore);

const mockedStore = reactive({
  commit: vi.fn()
});

describe('Given the `ProfileIcon` component', () => {
  const render = () => shallowMount(ProfileIcon);
  let wrapper: VueWrapper<InstanceType<typeof ProfileIcon>>;

  beforeAll(() => {
    mockedUseStore.mockReturnValue(mockedStore);
  });

  describe('when it is rendered', () => {
    beforeEach(() => {
      wrapper = render();
    });

    it('should match the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    describe('and when the `View Profile` link is clicked', () => {
      beforeEach(() => {
        // wrapper.find('[data-test="signout"]').trigger('click');
        wrapper.vm.toggleModal();
      });

      it('should emit the correct event', () => {
        expect(mockedStore.commit).toHaveBeenCalledWith('user/toggleModal');
      });

      it('should update the component state', () => {
        expect(wrapper.vm.isDropdownOpen).toBe(true);
      });
    });

    describe('and when the `Sign Out` link is clicked', () => {
      beforeEach(() => {
        // wrapper.find('[data-test="signout"]').trigger('click');
        wrapper.vm.signOut();
      });

      it('should emit the correct event', () => {
        expect(wrapper.emitted('signout')).toStrictEqual([[]]);
      });
    });
  });
});
