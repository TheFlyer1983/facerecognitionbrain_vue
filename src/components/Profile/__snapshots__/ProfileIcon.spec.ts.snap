// Vitest Snapshot v1

exports[`Given the \`ProfileIcon\` component > when it is rendered > should match the snapshot 1`] = `
<section
  class="dropdown"
  data-v-462d6e3c=""
>
  <span
    class="dropdown-button pointer"
    data-v-462d6e3c=""
  >
    <img
      alt="avatar"
      class="dropdown-button-image"
      data-v-462d6e3c=""
      src="http://tachyons.io/img/logo.jpg"
    />
  </span>
  <!--v-if-->
</section>
`;
